src/main/java/htsjdk/samtools/seekablestream/SeekableStreamFactory.java

```diff
@@ -61,9 +72,28 @@ public static ISeekableStreamFactory getInstance(){
      * Does this path point to a regular file on disk and not something like a URL?
      * @param path the path to test
      * @return true if the path is to a file on disk
+     * @deprecated this method is simplistic and no longer particularly useful since IOPath allows similar access to
+     * various non-file data sources, internal use has been replaced with {@link #isBeingHandledByLegacyUrlSupport(String)}
      */
+    @Deprecated
     public static boolean isFilePath(final String path) {
-        return ! ( path.startsWith("http:") || path.startsWith("https:") || path.startsWith("ftp:") );
+        return !canBeHandledByLegacyUrlSupport(path);
+    }
+
+    /**
+     * is this path being handled by one of the legacy SeekableStream types (http(s) / ftp)
+     *
+     * @param path a path to check
+     * @return if the path is not being handled by a FileSystemProvider and it can be read by legacy streams
+     */
+    public static boolean isBeingHandledByLegacyUrlSupport(final String path){
+        return !new HtsPath(path).hasFileSystemProvider()  //if we have a provider for it that's what we'll use
+                && canBeHandledByLegacyUrlSupport(path); // otherwise we fall back to the special handlers
+    }
+
+    //is this onee of the url types that has legacy htsjdk support built in?
+    public static boolean canBeHandledByLegacyUrlSupport(final String path) {
+        return URL_SCHEMES_WITH_LEGACY_SUPPORT.stream().anyMatch(path::startsWith);
```

Comment:
Shouldn't you be checking that the path starts with "http://" (for example) instead of "http"? What if there's a file named "http"?